include/logging/log_core.h

```diff
@@ -14,6 +14,40 @@
 extern "C" {
 #endif
 
+/* MOJ KOD */
+
+#include <zephyr.h>
+#include <device.h>
+#include <gpio.h>
+#include <hal/nrf_gpio.h>
+
+#if (CONFIG_LOG_FRONTEND_PORT_NUMBER == 0)
+#define LOG_FRONTEND_GPIO_REG_PTR NRF_P0
+#elif (CONFIG_LOG_FRONTEND_PORT_NUMBER == 1)
+#define LOG_FRONTEND_GPIO_REG_PTR NRF_P1
+#endif
+
+struct LOG_FRONTEND_PIN_NUMBERS
+{
+	uint8_t BIT0;
+	uint8_t BIT1;
+	uint8_t BIT2;
+	uint8_t BIT3;
+	uint8_t BIT4;
+	uint8_t BIT5;
+	uint8_t BIT6;
+	uint8_t BIT7;
+	uint8_t CLK;
+};
+
+uint32_t create_mask_gpio(struct LOG_FRONTEND_PIN_NUMBERS * pin_nums, uint8_t message);
+uint32_t create_mask_gpio_of_pins(struct LOG_FRONTEND_PIN_NUMBERS * pin_nums);
+void send_message_gpio_8b(uint8_t message, struct LOG_FRONTEND_PIN_NUMBERS * pin_nums, uint32_t mask_of_pins);
```

Comment:
this functions should be moved to log_frontend_nrf_gpio.c, they are internal to this particular frontend.
Here you should rather define functions like nrf_log_frontend_0(...),...